### PR TITLE
[MINOR][TESTS] Ignore GitHub Action and AppVeyor file changes in testing

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to ignore `GitHub Action` and `AppVeyor` file changes. When we touch these files, Jenkins job should not trigger a full testing.

### Why are the changes needed?

Currently, these files are categorized to `root` and trigger the full testing and ends up wasting the Jenkins resources.
- https://github.com/apache/spark/pull/26555
```
[info] Using build tool sbt with Hadoop profile hadoop2.7 under environment amplab_jenkins
From https://github.com/apache/spark
 * [new branch]      master     -> master
[info] Found the following changed modules: sparkr, root
[info] Setup the following environment variables for tests: 
```

### Does this PR introduce any user-facing change?

No. (Jenkins testing only).

### How was this patch tested?

Manually.
```
$ dev/run-tests.py -h -v
...
Trying:
    [x.name for x in determine_modules_for_files([".github/workflows/master.yml", "appveyor.xml"])]
Expecting:
    []
...
```

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Hi, @srowen and @HyukjinKwon .
This is helpful when we investigate AppVeyor or GitHub Action only issue.
This will reduce the waste of our Jenkins resources.
Could you review this PR?

**Comment by dongjoon-hyun** [comment]
As I mentioned in #26133 , AppVeyor is the only way to run SparkR arrow test. I'll merge this because we need more job triggers to upgrade SparkR Arrow.

**Comment by dongjoon-hyun** [comment]
Merged to master.

File: dev/run-tests.py [hunk 1]
```diff
@@ -43,15 +43,20 @@ def determine_modules_for_files(filenames):
...
     If a file is not associated with a more specific submodule, then this method will consider that
-    file to belong to the 'root' module.
+    file to belong to the 'root' module. GitHub Action and Appveyor files are ignored.
 
     >>> sorted(x.name for x in determine_modules_for_files(["python/pyspark/a.py", "sql/core/foo"]))
     ['pyspark-core', 'sql']
     >>> [x.name for x in determine_modules_for_files(["file_not_matched_by_any_subproject"])]
     ['root']
+    >>> [x.name for x in determine_modules_for_files( \
+            [".github/workflows/master.yml", "appveyor.yml"])]
```
Comment (HyukjinKwon) [inline dev/run-tests.py]: no big deal but it is usually:

```python
    >>> [x.name for x in determine_modules_for_files(
    ...     [".github/workflows/master.yml", "appveyor.yml"])]
```

**Review by HyukjinKwon — COMMENTED** [review]
Looks good to me.

**Pull request closed without merging**